### PR TITLE
add connector blueprint for cohere embedding models in bedrock

### PR DESCRIPTION
**Pull request by Zhangxunmt**
### Description
Blueprints for cohere embedding models in bedrock.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ylwu-amzn** [comment]
How about add to https://github.com/opensearch-project/ml-commons/blob/2.x/docs/remote_inference_blueprints/cohere_connector_embedding_blueprint.md

**Comment by Zhangxunmt** [comment]
> How about add to https://github.com/opensearch-project/ml-commons/blob/2.x/docs/remote_inference_blueprints/cohere_connector_embedding_blueprint.md, most part should be same ?

These two are pretty consistent with the current bedrock model blueprints examples. There are separate azure_openai and openai connectors. So why we want to merge them together? Cohere deployed inside Bedrock can be considered as bedrock models too.

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  "task_id": "rMormY8B8aiZvtEZIO_j",
+  "status": "CREATED",
+  "model_id": "lyjxwZABNrAVdFa9zrcZ"
+}
+```
+
+## 5. Test model inference
+
+```json
+POST /_plugins/_ml/models/rcormY8B8aiZvtEZIe89/_predict
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md:145]: Can you also try if this API can work ?  neural-search requires this API can work

```
POST /_plugins/_ml/_predict/text_embedding/<model_id>
{
  "text_docs":[ "today is sunny", "today is sunny"],
  "return_number": true,
  "target_response": ["sentence_embedding"]
}
```

Comment (Zhangxunmt) [reply docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md:145]: Yes this is also verified too. It invokes successfully.

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+}
+```
+
+If using the AWS Opensearch Service, you can provide an IAM role arn that allows access to the bedrock service.
+Refer to this [AWS doc](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ml-amazon-connector.html)
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "Amazon Bedrock",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md:64]:
```suggestion
    "name": "Amazon Bedrock Cohere embed-english-v3",
```

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-multilingual-v3  _blueprint.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+}
+```
+
+If using the AWS Opensearch Service, you can provide an IAM role arn that allows access to the bedrock service.
+Refer to this [AWS doc](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/ml-amazon-connector.html)
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "Amazon Bedrock",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-multilingual-v3  _blueprint.md:64]:
```suggestion
    "name": "Amazon Bedrock Connector: Cohere embed-multilingual-v3",
```

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  "connector_id": "hijwwZABNrAVdFa9prf7"
+}
+```
+
+## 3. Create model group:
+
+```json
+POST /_plugins/_ml/model_groups/_register
+{
+    "name": "remote_model_group_claude3",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md:107]: This is not claude3

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
+# Bedrock connector blueprint example for Cohere embed-english-v3 model
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md:1]: Add AWS doc link https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed.html?

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+    "description": "Test connector for Amazon Bedrock",
+    "version": 1,
+    "protocol": "aws_sigv4",
+    "credential": {
+        "roleArn": "<PLEASE ADD YOUR AWS ROLE ARN HERE>"
+    },
+    "parameters": {
+      "region": "<PLEASE ADD YOUR AWS REGION HERE>",
+      "service_name": "bedrock",
+      "truncate": "<NONE|START|END>",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md:74]: Add some explanation for `truncate` and `input_type`?

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-multilingual-v3  _blueprint.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+  "connector_id": "ISj-wZABNrAVdFa9cLju"
+}
+```
+
+## 3. Create model group:
+
+```json
+POST /_plugins/_ml/model_groups/_register
+{
+    "name": "remote_model_group_claude3",
```
Comment (ylwu-amzn) [inline docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-multilingual-v3  _blueprint.md:107]: not claude3

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md (hunk 6 above)
Comment (Zhangxunmt) [reply docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md:74]: Added the AWS doc link to explain all parameters in the latest commit.

File: docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md (hunk 4 above)
Comment (Zhangxunmt) [reply docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-english-v3_blueprint.md]: updated to a different name.

**Review by ylwu-amzn — APPROVED** [review]

**Review by b4sjoo — APPROVED** [review]

**Pull request merged**